ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,196 @@
+}
+
+void op::v7::DFT::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v7_DFT_validate_and_infer_types);
+
+    size_t num_of_inputs = get_input_size();
+
+    NODE_VALIDATION_CHECK(
+        this, num_of_inputs == 2 || num_of_inputs == 3, "DFT must have 2 or 3 inputs.");
+
+    element::Type input_et = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_et == element::f32 || input_et == element::f16 ||
+                              input_et == element::bf16,
+                          "Input element type must be f32, f16, or bf16");
+
+    element::Type axes_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          axes_et == element::i64 || axes_et == element::i32 ||
+                              axes_et == element::u32 || axes_et == element::u64,
+                          "Axes element type must be i32, i64, u32 or u64");
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    PartialShape axes_shape = PartialShape(get_input_partial_shape(1));
+    PartialShape output_shape = input_shape;
+
+    if (input_shape.rank().is_dynamic() || axes_shape.rank().is_dynamic() ||
+        !is_type<op::Constant>(input_value(1).get_node()))
+    {
+        set_output_type(0, get_input_element_type(0), output_shape);
+        return;
+    }
+
+    const auto input_rank = input_shape.rank().get_length();
+
+    NODE_VALIDATION_CHECK(this,
+                          axes_shape.rank().get_length() == 1,
+                          "Axes input must be 1D tensor. Got axes input rank: ",
+                          axes_shape.rank().get_length());
+
+    auto last_dim_with_two = input_shape[input_rank - 1] & Dimension(2);
+    NODE_VALIDATION_CHECK(this,
+                          !last_dim_with_two.get_interval().empty(),
+                          "The last dimension of input data must be 2. Got: ",
+                          input_shape[input_rank - 1]);
+
+    const auto& const_axes = get_constant_from_source(input_value(1));
+    auto axes = const_axes->cast_vector<int64_t>();
+
+    for (int64_t& axis : axes)
+    {
+        if (axis < 0)
+        {
+            axis += input_rank - 1;
+        }
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          input_rank >= axes.size() + 1,
+                          "The input rank must be greater than number of DFT axes. Got "
+                          "input rank: ",
+                          input_rank,
+                          ", number of axes: ",
+                          axes.size());
+
+    AxisVector axes_vector;
+    AxisSet axes_set;
+    for (const int64_t axis : axes)
+    {
+        axes_vector.push_back(static_cast<size_t>(axis));
+        axes_set.insert(static_cast<size_t>(axis));
+    }
+
+    NODE_VALIDATION_CHECK(
+        this, axes_vector.size() == axes.size(), "DFT axes must be unique. Got: ", axes_vector);
```

Comment:
Probably here should be axes_vector.size() == axes_set.size() ? And why do we need axes_vector? Сan we compare axes_set.size() == axes.size() instead?

Reply:
Yes, here `should be axes_vector.size() == axes_set.size()`, but we can use `axes_set.size() == axes.size()`. `axes_vector` is needed, because there are no `std::ostream& operator<<(std::ostream& stream, const std::vector<int64_t> v)`, but there are similar operator for `AxisVector`. This operator is used in lines 142-148.